File: Utility/GCD_LCM.cpp
#include "../_CppTemplate.cpp"

ll GCD(ll a, ll b) {
  if (a == 0 || b == 0) return 0;
  while (a != 0 && b != 0) {
    if (a > b)
      a = a % b;
    else
      b = b % a;
  }
  return max(a, b);
}

ll LCM(ll a, ll b) {
  if (a == 0 || b == 0) return 0;
  return a * b / GCD(a, b);
}

File: Utility_Ruby/GCD_LCM.rb

def GCD(m, n)
	return 0 if m == 0 || n == 0
	while m != 0 && n != 0
		if m > n then
			m %= n
		else
			n %= m
		end
	end
	return [m, n].max
end

def LCM(m, n)
	return 0 if m == 0 || n == 0
	return m * n / GCD(m, n)
end

